src/Microsoft.Azure.WebJobs.Extensions.Storage/StorageAccountProvider.cs

```diff
@@ -40,10 +41,15 @@ public virtual StorageAccount Get(string name)
             if (connectionString == null)
             {
                 // Not found
-                throw new InvalidOperationException($"Storage account connection string '{name}' does not exist. Make sure that it is a defined App Setting.");
+                throw new InvalidOperationException($"Storage account connection string '{IConfigurationExtensions.GetPrefixedConnectionStringName(name)}' does not exist. Make sure that it is a defined App Setting.");
             }
 
-            return StorageAccount.NewFromConnectionString(connectionString);
```

Comment:
Where else is `NewFromConnectionString()` used? Should this logic be moved there so all callers can benefit?

Reply:
The only other place i see it being used is in tests
- https://github.com/Azure/azure-webjobs-sdk/blob/dev/test/Microsoft.Azure.Webjobs.Extensions.Storage.UnitTests/HostCallTests.cs#L963
- https://github.com/Azure/azure-webjobs-sdk/blob/dev/test/Microsoft.Azure.WebJobs.Host.EndToEndTests/MultipleStorageAccountsEndToEndTests.cs#L267

Reply:
Is it easy to move the logic into there? Just for any future usage as well.

Reply:
@brettsam do you mean move the validation logic into NewFromConnectionString method or into the tests. I think i can move the logic to NewFromConnectionString method. But for that would have to introduce a new  parameter since the method does not know about the source of the connections string, only the connection string value.

Reply:
Oh, to improve the error message, of course. And unfortunately NewFromConnectionString is public so we can't delete it, etiher. Oh well, yeah I guess this works :-)